src/hooks/UrqlUseSubscriptionWithHandler.re

```diff
@@ -0,0 +1,72 @@
+[@bs.deriving abstract]
+type useSubscriptionWithHandlerArgs = {
+  query: string,
+  [@bs.optional]
+  variables: Js.Json.t,
+};
+
+[@bs.deriving abstract]
+type useSubscriptionWithHandlerResponseJs('acc) = {
+  fetching: bool,
+  [@bs.optional]
+  data: 'acc,
+  [@bs.optional]
+  error: UrqlCombinedError.t,
+};
+
+type useSubscriptionWithHandlerResponse('acc) = {
+  fetching: bool,
+  data: option('acc),
+  error: option(UrqlCombinedError.t),
+  response: UrqlTypes.response('acc),
+};
+
+[@bs.module "urql"]
+external useSubscriptionWithHandlerJs:
+  (
+    useSubscriptionWithHandlerArgs,
+    UrqlTypes.parsedHandler('acc, 'response)
+  ) =>
+  array(useSubscriptionWithHandlerResponseJs('acc)) =
+  "useSubscription";
+
+let useSubscriptionResponseToRecordWithHandler =
+    (result: useSubscriptionWithHandlerResponseJs('acc)) => {
+  let data = result->dataGet;
+  let error = result->errorGet;
+  let fetching = result->fetchingGet;
+
+  let response =
+    switch (fetching, data, error) {
+    | (true, None, _) => UrqlTypes.Fetching
+    | (true, Some(d), _) => Data(d)
```

Comment:
So `urql` has a strange case where `fetching` is never set to `false` on subscriptions (I've pinged folks internally to ensure that's intended behavior). To handle this, I needed to add this extra case to this pattern matching to check if we are receiving data even as `fetching` remains true.